### ISSUE TITLE
"error: cannot find macro `invoke_sqlite` in this scope" when trying to create shared library

### ISSUE THREAD

**Issue by fivestones** [vlcn-io/cr-sqlite#152]
I am trying to follow along with the Getting Started section of the docs, in the "existing build" section.
I did `git clone git@github.com:vlcn-io/cr-sqlite.git`, then `cd cr-sqlite/core`, and then `make loadable` which immediately errored, as the core/rs/sqlite-rs-embedded folder was empty. I had to do `cd rs` and then `git clone git@github.com:vlcn-io/sqlite-rs-embedded.git`, which fixed that problem. Going back to the core folder, I again did `make loadable` which looked like it was working, until it got to compling sqlite3_capi, where it produced 54 errors, almost all of which were the same: ``cannot find macro `invoke_sqlite` in this scope``.

I'm not sure where to go from here. Searching the source for invoke_sqlite finds the function, but I'm not sure what the `make loadable` needs for this to work. 

I'm on  MacOS if it makes a difference.

Any help would be appreciated.

**Comment by tantaman** [comment]
Thanks for reporting. Forgot to update the docs after introducing some rust dependencies.

If you want to build from source (rather than use an existing package) you'll need to do a recursive checkout.

```
git clone --recurse-submodules git@github.com:vlcn-io/cr-sqlite.git
```

or, if you've already cloned,

```
git submodule update --init
```

lmk how that goes. I'll update the instructions and retry a fresh clone here to be sure nothing has changed.

**Comment by tantaman** [comment]
also, make sure you have the rust nightly toolchain installed:

```
rustup toolchain install nightly
```

**Comment by tantaman** [comment]
Updated the docs to say:

```bash
rustup toolchain install nightly # make sure you have the rust nightly toolchain
git clone --recurse-submodules git@github.com:vlcn-io/cr-sqlite.git
cd cr-sqlite/core
make loadable
```


---

> I had to do cd rs and then git clone git@github.com:vlcn-io/sqlite-rs-embedded.git

This didn't work fully since the main repository points to a specific version of the sub repos. The latest version of `sqlite-rs-embedded` can't be included until a few other prs are merged.

**Comment by fivestones** [comment]
Great! Thanks, that was very helpful.

One more thing that tripped me up after I followed these new instructions, I was getting an error ``[`#![feature]` may not be used on the stable release channel``. After some searching on google I figured out that to actually use the nightly toolchain, I first had to do `rustup default nightly`. Then the error was solved. This is probably obvious to most people who are used to using rust, but it might help people like me if you included that in the Getting Started docs too.
